src/qvi/qvi.ts

```diff
@@ -1,51 +1,100 @@
-import {SignifyClient} from "signify-ts";
-import {DONE, NAME, RESPONSE} from "../operations";
+import { SignifyClient } from "signify-ts";
+import { waitKeriaOp } from "../operations";
+import { getQviRegistry } from "../registry";
 
 
-// import LE from "../schema/legal-entity-vLEI-credential.json";
+import LE from "../schema/legal-entity-vLEI-credential.json";
+import ECR from "../schema/legal-entity-engagement-context-role-vLEI-credential.json";
+import OOR from "../schema/legal-entity-engagement-context-role-vLEI-credential.json";
+
+
+import {ECRVLEICredential, LEVLEICredential, LE_VLEI_EDGES, ECR_AUTH_EDGES, ECR_LE_EDGES, OOR_AUTH_EDGES, OORVLEICredential} from '../interfaces'
+
 export class QVI {
-    public readonly SCHEMA_CACHE: { LEGAL_ENTITY: any; };
+    private client: SignifyClient;
     private readonly alias: string;
     private readonly registryName: string;
     private readonly registryNonce: string;
     private registry: any;
+    private readonly SCHEMA_CACHE: { LEGAL_ENTITY: any; ENGAGEMENT_CONTEXT_ROLE: any; OFFICIAL_ORGANIZATIONAL_ROLE: any; };
 
     constructor(alias: string, registryName: string, registryNonce: string) {
         this.alias = alias;
         this.registryName = registryName;
         this.registryNonce = registryNonce;
-        let LE = {};
         this.SCHEMA_CACHE = {
             LEGAL_ENTITY: {
                 LE: LE,
-                RULES: [
-                    // @ts-ignore
-                    JSON.stringify(LE.properties.r.oneOf[1].properties.usageDisclaimer.properties.l.const),
-                    // @ts-ignore
-                    JSON.stringify(LE.properties.r.oneOf[1].properties.issuanceDisclaimer.properties.l.const),
-                ],
+                RULES: {
+                    "d": "EDIai3Wkd-Z_4cezz9nYEcCK3KNH5saLvZoS_84JL6NU",
+                    "usageDisclaimer": {
+                        // @ts-ignore
```

Comment:
this is ignored because I'm not sure it works...

Reply:
to remove this error we can either remove  `"strict":true` from the tsconfig or do 
`LE?.properties?.r?.oneOf[1]?.properties?`with the question marks. I would prefer the second option